webapp/src/pages/LoginPage/LoginPage.tsx

```diff
@@ -38,10 +39,14 @@ export default function LoginPage() {
       const authUser = await authApi.authControllerLoginLocalUser({
         loginUserDto: { username, password },
       });
-      const { gOwner, gAdmin, gBanned } =
-        await authApi.authControllerRetrieveAuthUserWithRoles();
-      setAuthUser({ ...authUser, gOwner, gAdmin, gBanned });
-      navigate(DEFAULT_LOGIN_REDIRECT_URL, { replace: true });
+      if (authUser.isTwoFactorAuthenticationEnabled) {
+        navigate(TWO_FACTOR_AUTH_VALIDATE_URL, { replace: true });
```

Comment:
what happens if an admin is 2fa enabled? Will they not have their authUser set to have admin rights? (I hit the bug I've reported while investigating this)

Reply:
I think we deal with that case in the `ValidateTwoFactorAuthPage` where we set the `authUser` and the permissions.